### PR TITLE
(BE - Infrastructure): Upgraded aws-cdk-lib Python package and Python…

### PR DESCRIPTION
**Pull request by samzofkie**
… runtime version used in the CDK code. Also updated BE README to describe difference in deployment process. 

This has been deployed successfully 🤘

### CODE REVIEW

**Comment by MatthewMSaucedo** [comment]
How was this tested?

File: backend/infra/infra/cw_core_stack.py [hunk 1]
```diff
@@ -386,25 +386,25 @@ def create_cw_validator_lambda(self, jwt_secret, jwt_exp_minutes):
...
             id=f"{self.stack_env}-cw-validator-lambda",
-            runtime=lambdaFx.Runtime.PYTHON_3_7,
+            runtime=lambdaFx.Runtime.PYTHON_3_12,
             handler="index.handler",
             role=validator_role,
             code=lambdaFx.Code.from_asset("./lambda/validator/"),
             description="CarWorld Validator Lambda, to authenticate API requests",
             environment={"jwtSecret": jwt_secret, "jwtExpMinutes": jwt_exp_minutes},
-            layers=[jwt_lambda_layer, jwt_dependency_lambda_layer],
+            layers=[jwt_lambda_layer], # jwt_dependency_lambda_layer],
```
Comment (MatthewMSaucedo) [inline backend/infra/infra/cw_core_stack.py]: If we are removing this, we shouldn't leave the comment

File: backend/infra/infra/cw_core_stack.py [hunk 2]
```diff
@@ -417,7 +417,8 @@ def create_cw_api_gw(self):
             scope=self,
             id=f"{self.stack_env}-cw-auth-api",
             description="CarWorld API Gateway",
-            cors_preflight=apigw.CorsPreflightOptions(
+            #TODO: look into this
```
Comment (MatthewMSaucedo) [inline backend/infra/infra/cw_core_stack.py]: Nit: Weird line spacing introduced?

Request: Can we improve the this comment with more context if you'd like to leave it in?

**Review by MatthewMSaucedo — CHANGES_REQUESTED** [review]
Couple comments for you to look at

**Comment by samzofkie** [comment]
> How was this tested?

Only tested by deploying and running the frontend dev server. Auth works, commerce does not ATM. Did you have any thing else in mind test-wise?

File: backend/infra/infra/cw_core_stack.py (hunk 2 above)
Comment (samzofkie) [reply backend/infra/infra/cw_core_stack.py]: Just edited the first commit in this PR to fix these two issues!

**Review by MatthewMSaucedo — DISMISSED** [review]

**Comment by samzofkie** [comment]
Ok now this has been deployed and the auth, validator, and commerce lambdas all work for at least opening the Stripe iframe to checkout as a guest. The `backend/README` details how to deploy. I ended up rolling back from Python 3.12 to 3.11, only because it's important to run the `pip` commands with the same version of Python that the lambda's runtime will use, and the Linux distro I'm working from right now doesn't support 3.12 yet and I just didn't feel like installing 3.12 ❤️ (I added a comment in the `README` describing the unintuitive error the validator lambda gave me because the Python versions didn't match).

**Review by MatthewMSaucedo — APPROVED** [review]

**Pull request merged**